TestRunners/PassTestRunner/run_test.py

```diff
@@ -0,0 +1,77 @@
+import os
+import sys
+
+if os.path.split(os.getcwd())[1] != "build":
+    print(f"WARNING: not in build dir; cwd is {os.getcwd()}")
+    sys.exit(1)
+
+sys.path.insert(0, '..')
+from TestRunners.FlagsManager import FlagsManager
+from PassTestFilesManager import PassTestFilesManager
+from PassTestRunner import PassTestRunner
+
+
+if __name__ == "__main__":
+    '''
+    Usage examples: 
+    python3 ../run_pt.py [file name] [flags]
+    python3 ../run_pt.py -h (to get help) 
+
+    python3 ../run_pt.py simple_ptr_test
+    python3 ../run_pt.py simple_ptr_test --no-make --no-build-ir
+    python3 ../run_pt.py simple_layer_test --only-run-for test1.c
+    python3 ../run_pt.py simple_layer_test --only-run-for layer/test1_again.c
+    '''
+
+    flags_manager = FlagsManager(
+        "python3 ../run_pt.py [required test folder name] [optional flags]")
+    flags_manager.add_flag(
+        "h", "help",
+        "Show help"
+    )
+
+    flags_manager.add_flag(
+        "m", "no-make",
+        "Does not call 'make' command",
+        lambda pass_test_files_manager: pass_test_files_manager.toggle_make_call()
+    )
+
+    flags_manager.add_flag(
+        "b", "no-build-ir",
+        "Does not build IR for any file. Combined with --only-build-ir-for may result in undesirable output as these commands are opposites. ",
+        lambda pass_test_files_manager: [file.toggle_ir_generation(
+        ) for file in pass_test_files_manager.get_all_files()]
+    )
+
+    flags_manager.add_flag(
+        "f", "no-build-ir-for",
+        "Does not build IR for the specified c file. Can be called multiple times. c file must be specified in the succeeding argument.",
+        lambda pass_test_files_manager, file_name: pass_test_files_manager.get_file(
+            file_name).toggle_ir_generation()
+
+    )
+    flags_manager.add_flag(
+        "u", "only-build-ir-for",
+        "Only builds ir for specified c file. c file must be specified in the succeeding argument. Note that c files will still be ran unless otherwise disallowed (use -n). Combined with --no-build-ir may result in undesirable output as these commands are opposites.",
```

Comment:
typo "still be ran" should be "still be run", I think (run is an irregular verb though so I might actually be wrong about this - English class!)

Reply:
I'd prefer that combining with `--no-build-ir` leads to an exception if its non-sensical (in both the handling of this command and that one)

Reply:
Grammarly agrees with you that it'd be "run". But I've been using the phrase "undesirable output" to say that, because of how the CLT's are implemented, combining "opposite" flags may run the code exactly opposite to how the user expects

Foe example, let's say we call the test with flags `--no-run-for file1.c --only-run-for file1.c`, and say our test directory has files file1.c, file2.c, and file3.c. 
Each test file has an initial condition `TEST_WILL_RUN` set to true (defined in `TestFile.py`). The only way we can change it is by calling `TestFile.toggle_test_running`
As the flags are parsed from left to right, the following is executed:
`--no-run-for file1.c` => calls file1.toggle_test_running(), which now equals False
`--only-run-for file1.c` => calls toggle_test.running() for all files except `file1.c`. 
This results in no test running. 

I'll make it so an exception is thrown if "opposites" are called